syft/pkg/cataloger/catalog.go

```diff
@@ -168,14 +175,10 @@ func Catalog(resolver source.FileResolver, release *linux.Release, parallelism i
 
 	allRelationships = append(allRelationships, pkg.NewRelationships(catalog)...)
 
-	if errs != nil {
-		return nil, nil, errs
-	}
-
 	filesProcessed.SetCompleted()
 	packagesDiscovered.SetCompleted()
```

Comment:
The fact that this function was returning _before_ the `.SetCompleted()` calls below resulted in a hung TUI for any error that occurs.

Reply:
oh duh! man, I sent you down the hard path too soon.

Reply:
actually, can you add a little more defensiveness to this (protecting against future changes)? Can you add something to the affect of:
```golang
defer filesProcessed.SetCompleted()
defer packagesDiscovered.SetCompleted()
```

where these are created with `newMonitor()`?

Reply:
Good call -- done!